### PR TITLE
RFC: Update template

### PR DESCRIPTION
**Pull request by Scarysize**
#33

### CODE REVIEW

File: templates/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        query: {
+          presets: ['env'],
+          plugins: ['transform-object-rest-spread']
+        }
+      },
+      {
+        test: /\.pug$/,
```
Comment (Scarysize) [inline templates/webpack.config.js]: do we actually need pug templating or would a simple `index.html` file suffice?

File: templates/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  devtool: isProduction ? false : 'source-map',
+  module: {
+    loaders: [
+      {
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        query: {
+          presets: ['env'],
+          plugins: ['transform-object-rest-spread']
```
Comment (Scarysize) [inline templates/webpack.config.js]: debatable.

File: templates/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  },
+  devServer: {
+    contentBase: './build',
+    port: 8000
+  },
+  devtool: isProduction ? false : 'source-map',
+  module: {
+    loaders: [
+      {
+        test: /\.jsx?$/,
```
Comment (Scarysize) [inline templates/webpack.config.js]: should we add `react` preset by default?

File: templates/index-js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import '../styles/index.styl';
```
Comment (Scarysize) [inline templates/index-js]: This isn't css-modules. Just a top-level import for webpack.

File: templates/webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        test: /\.pug$/,
+        exclude: /node_modules/,
+        loader: 'pug-loader'
+      },
+      {
+        test: /\.styl$/i,
+        exclude: /node_modules/,
+        // always extract the css into a file
+        use: ExtractTextPlugin.extract({
+          use: ['css-loader', 'autoprefixer-loader', 'stylus-loader']
```
Comment (Scarysize) [inline templates/webpack.config.js]: autoprefixer loader is deprecated, recommendation is `postcss-loader`.

File: setup.sh [hunk 6]
```diff
@@ -15,21 +15,25 @@ export config_loglevel
 TEMPLATES_BASE_URL="http://ubilabs.github.io/project-template/templates/"
 
 NPM_MODULES=(
-  autoprefixer-stylus
+  autoprefixer-loader
   babel-core
-  babelify
-  browserify
```
Comment (usefulthink) [inline setup.sh]: yay!

File: templates/package.json [hunk 7]
```diff
@@ -6,6 +6,18 @@
...
+  },
   "scripts": {
+    "build": "npm run clean && NODE_ENV=production webpack -p",
+    "changelog": "conventional-changelog-generator && $EDITOR -w CHANGELOG.md",
+    "clean": "rm -rf build/*",
+    "deploy": "gcloud-storage-upload -s '{{project-name}}' -p build",
+    "postversion": "npm run changelog && git add CHANGELOG.md && git commit --amend --no-edit",
+    "preversion": "npm test",
+    "start": "npm run clean && webpack-dev-server",
+    "test": "eslint webpack.config.js src/scripts"
```
Comment (usefulthink) [inline templates/package.json]: do we want prettier? :D

File: templates/webpack.config.js (hunk 3 above)
Comment (usefulthink) [inline templates/webpack.config.js]: I'd say we should maybe have another project-template for react/preact. Also, what about preact-cli and create-react-app? They are customizable, aren't they?

Comment (Scarysize) [inline templates/webpack.config.js]: I would agree. It's trivial to add react/preact support to the webpack.config.

File: templates/package.json (hunk 7 above)
Comment (Scarysize) [inline templates/package.json]: Maybe we should have that discussion separately 😉

File: templates/webpack.config.js (hunk 3 above)
Comment (yfr) [inline templates/webpack.config.js]: What about `js` files?

File: templates/webpack.config.js (hunk 2 above)
Comment (yfr) [inline templates/webpack.config.js]: Would not add specials like this. Just the minimum required.

File: templates/webpack.config.js (hunk 1 above)
Comment (yfr) [inline templates/webpack.config.js]: For starters, I would go with the webpack html plugin.

File: templates/package.json [hunk 8]
```diff
@@ -6,6 +6,18 @@
     "type": "git",
     "url": "https://github.com/ubilabs/{{project-name}}.git"
   },
+  "precommit": {
+    "run": "test",
+    "template": ".gitcommit.template"
+  },
   "scripts": {
+    "build": "npm run clean && NODE_ENV=production webpack -p",
+    "changelog": "conventional-changelog-generator && $EDITOR -w CHANGELOG.md",
```
Comment (yfr) [inline templates/package.json]: Does that work with the npm scripts? Do the scripts wait for the editor to finish? And is every editor able to do this?

File: templates/webpack.config.js (hunk 2 above)
Comment (Scarysize) [inline templates/webpack.config.js]: I'd argue that we use this over `Object.assign` in most of the current projects, especially with `redux` in the stack.

File: templates/package.json (hunk 8 above)
Comment (Scarysize) [inline templates/package.json]: Hm probably only atom. Remove completely? Is there a unix equivalent for `-w`?

File: templates/webpack.config.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        exclude: /node_modules/,
+        // always extract the css into a file
+        use: ExtractTextPlugin.extract({
+          use: [
+            'css-loader',
+            {
+              loader: 'postcss-loader',
+              options: {
+                plugins: isProduction ? [require('autoprefixer')] : null,
+                sourceMap: true
```
Comment (yfr) [inline templates/webpack.config.js]: Why prefixer only in prod mode and source map everytime?

**Pull request merged**